### ISSUE TITLE
json_extract doesnt extract keys with double quotes

### ISSUE THREAD

**Issue by leonpanokarren** [prestodb/presto#7499]
A JSON column in my source table has the structure
```json
 "parameterValues": {
    "DE:Design and/or Copy - Deliverables Requested": "Presentation Design/Copy",
  },
```
I am unable to JSON_EXTRACT this successfully. Tried

```sql
json_format(json_extract(object_json, '$["parameterValues"]["DE:Design and/or Copy - \"Other\" Deliverables Requested Description"]'))
```
and this

```sql
json_format(json_extract(object_json, '$["parameterValues"]["DE:Design and/or Copy - '"Other"" Deliverables Requested Description"]'))`
```
How do I get around double quotes in keys? I thought backslashes effectively escaped quotes.

**Comment by electrum** [comment]
We should support escaped backslashes via `\"` in JSON paths. This should be an easy change to `JsonPathTokenizer`.

**Comment by electrum** [comment]
Unfortunately, this will be a backwards incompatible change because backslash will now need to be escaped. We should only allow a backslash to be followed by another backslash or a double quote.

**Comment by raghavsethi** [comment]
Merged #7509
